api/s2n.h

```diff
@@ -3391,6 +3391,27 @@ S2N_API int s2n_offered_early_data_reject(struct s2n_offered_early_data *early_d
  */
 S2N_API int s2n_offered_early_data_accept(struct s2n_offered_early_data *early_data);
 
+/**
+ * Retrieves the list of supported groups configured by the security policy associated with `config`.
+ *
+ * IANA values for each of the supported groups are written to the provided `groups` array, and `groups_count` is
+ * set to the number of written supported groups.
+ *
+ * `groups_count_max` should be set to the maximum capacity of the `groups` array. If `groups_count_max` is less
+ * than the number of supported groups configured by the security policy, this function will error.
+ *
+ * Note that this API retrieves the supported groups from a local security policy. To instead get the supported
```

Comment:
Yeah, the list of groups returned by the API is kind of confusing. I added a section to the API docs to try and explain this better.

Basically the API should return literally the groups on the policy, not necessarily what will be sent or considered compatible for a given connection (this is why it's a config API and not a connection API). For example, a server negotiating TLS 1.2 won't support TLS 1.3 kem groups, so these groups will not be considered compatible when receiving the extension. However, the API should still return these groups if they are in the security policy. Except - the API should not return groups that are in the security policy but not compatible with the libcrypto, since we won't even have the IANA values for these groups in this case.